crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -63,12 +69,21 @@ pub struct DWalletMPCManager {
     epoch_id: EpochId,
     weighted_threshold_access_structure: WeightedThresholdAccessStructure,
     pub(crate) validators_data_for_network_dkg: HashMap<PartyID, ValidatorDataForNetworkDKG>,
-    /// Sessions that are ready to advance when the next [`DWalletMPCDBMessage::PerformCryptographicComputations`]
-    /// message will be received.
-    /// We need this field to skip already completed rounds, & to use the same messages,
-    /// i.e. those that have been received until the first consensus round in which a quorum have reached.
-    /// We use this first consensus round to know that all validators advance with the exact same messages.
-    ready_to_advance: HashMap<ObjectID, DWalletMPCSession>,
+    /// A map of the pending cryptographic computation sessions.
+    /// This map is needed in order to remove a session that we received a quorum of messages for
+    /// its next round, so running the current, completed round is redundant.
+    pub(crate) pending_computation_map:
+        HashMap<DWalletMPCLocalComputationMetadata, DWalletMPCSession>,
+    /// The order of the [`pending_computation_map`]. Needed to process the computations in the order they were received.
+    pending_for_computation_order: VecDeque<DWalletMPCLocalComputationMetadata>,
+    /// The number of currently running cryptographic computations - i.e. computations we called [`rayon::spawn_fifo`] for,
+    /// but we didn't receive a completion message for.
+    pub(crate) currently_running_sessions_count: usize,
```

Comment:
why do we have this and also `active_sessions_counter`?
make the name clearer.
Maybe even move all cryptographic computation elements to a sub-struct.